src/renderer/components/Home.js

```diff
@@ -158,18 +164,26 @@ function TabPanel (props) {
 const useTabIndex = createPersistedState('currentView')
 
 export default function Home ({ onSelectLanguage }) {
-  const [dialog, setDialog] = React.useState()
+  const [dialog, setDialog] = React.useState(null)
+  const [error, setError] = React.useState(null)
   const [tabIndex, setTabIndex] = useTabIndex(0)
+  const [update, setUpdate] = useUpdater()
```

Comment:
Keeping update state in the `<Home>` component is going to cause a re-render of the whole app for each update event (e.g. progress events). This _may_ affect performance of the app when using it during an update. It could be resolved by memoizing the other components (territory and observation views) if they are not already, or it could be solved by placing the updater state in react context. It would be worth checking the performance / re-render cost with latest react devtools to see whether there is a performance cost.

Reply:
Good point, although I don't know enough about these details so I'll need to read up on it a bit